Mesh_3/include/CGAL/Mesh_3/C3T3_helpers.h

```diff
@@ -1514,7 +1514,8 @@ class C3T3_helpers
           c3t3.remove_from_complex(Facet(c,i));
 
         c->set_facet_surface_center(i, facet_surface_center_[old_i]);
-        c->set_facet_surface_center_index(i, surface_center_index_table_[old_i]);
+        const Facet mirror = c3t3.triangulation().mirror_facet(Facet(c, i));
+        mirror.first->set_facet_surface_center(mirror.second, facet_surface_center_[old_i]);
```

Comment:
I totally agree with the 2 added lines (and hardly believe they have never been there!)